secant/Features/Backup Flow/RecoveryPhraseDisplayStore.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  RecoveryPhraseDisplayStore.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/26/21.
+//
+
+import Foundation
+import ComposableArchitecture
+import UIKit
+
+struct BackupPhraseEnvironment {
+    var mainQueue: AnySchedulerOf<DispatchQueue>
+    var newPhrase: () -> Effect<RecoveryPhrase, AppError>
+}
+
+typealias RecoveryPhraseDisplayStore = Store<RecoveryPhraseDisplayState, RecoveryPhraseDisplayAction>
+
+struct RecoveryPhrase: Equatable {
+    struct Chunk: Hashable {
+        var startIndex: Int
+        var words: [String]
+    }
+
+    private let chunkSize = 6
+
+    let words: [String]
+
+    func toChunks() -> [Chunk] {
+        let chunks = words.count / chunkSize
+        return zip(0 ..< chunks, words.chunked(into: chunkSize)).map {
+            Chunk(startIndex: $0 * chunkSize + 1, words: $1)
+        }
+    }
+
+    func toString() -> String {
+        words.joined(separator: " ")
+    }
+}
+
+struct RecoveryPhraseDisplayState: Equatable {
+    var phrase: RecoveryPhrase?
+    var showCopyToBufferAlert = false
+}
+
+enum RecoveryPhraseDisplayAction: Equatable {
+    case createPhrase
+    case copyToBufferPressed
+    case finishedPressed
+    case phraseResponse(Result<RecoveryPhrase, AppError>)
+}
+
+typealias RecoveryPhraseDisplayReducer = Reducer<RecoveryPhraseDisplayState, RecoveryPhraseDisplayAction, BackupPhraseEnvironment>
+
+extension RecoveryPhraseDisplayReducer {
+    static let `default` = RecoveryPhraseDisplayReducer { state, action, environment in
+        switch action {
+        case .createPhrase:
+            return environment.newPhrase()
+                .receive(on: environment.mainQueue)
+                .catchToEffect(RecoveryPhraseDisplayAction.phraseResponse)
+        case .copyToBufferPressed:
+            guard let phrase = state.phrase?.toString() else { return .none }
+            UIPasteboard.general.string = phrase
```

Comment:
are you referring to line 61 `.catchToEffect(RecoveryPhraseDisplayAction.phraseResponse)`? 
Yes, MnemonicSwift should be wrapped to an Effect, Environment or whatever is used on TCA to wrap dependencies and inject them

Reply:
No I'm referring to the global usage of UIPasteboard

Reply:
Ohhh, yes! on the ECC Wallet I ended up doing some helper methods because is a feature that ends up being used in many places. 

https://github.com/zcash/zcash-ios-wallet/blob/master/wallet/wallet/Utils/PasteboardHelper.swift

Reply:
Marking as solved since we will work on issue https://github.com/zcash/secant-ios-wallet/issues/128